### ISSUE TITLE
LittleFS read fails, Core 6.1.13 and Home 3.4.4 with Espressif 32 v6.4.0

### ISSUE THREAD

**Issue by meakashrao** [platformio/platform-espressif32#1286]
Platform IO Serial Update
a) When I use Platform IO, and upload firmware.bin file using the upload button and littlefs.bin using the upload FileSystem Image, my ESP is able to read the LITTLEFS file.
b) When I upload using the OTA method by making the ESP go into Access Point Mode and use the files espota.py and esptool.py although both files are updated successfully my LittleFS read fails(mount doesn't but file read fails)
I tried to format the LITTLEFS partion using the function "LittleFS.format()" and it returns successful and uploaded the LITTLEFS.bin again and it still resulted in File not present and File read failed.
c) Then if I take the ESP, erase the flash contents, and repeat step 'a' things work again.
I am attaching a zip file with my files.
[ESP_OTA_LITTLEFS_Details.zip](https://github.com/platformio/platform-espressif32/files/13967106/ESP_OTA_LITTLEFS_Details.zip)

I have multiple sections where I send the Error Message using BLE a
The Error Messages are
a) it says "LITLFS-FileReadFail" in the function
void readFile(fs::FS &fs, const char * path), which basically just reads and serial prints contents of the 2 files in the data folder in Platform IO
b) Files not found
c) Reading fails again.

OTA upload Screenshots
Firmware File image
<img width="545" alt="image" src="https://github.com/platformio/platform-espressif32/assets/120377875/ea0aa6d9-8212-4582-a7cb-676e189c0370">

SPIFFS Image:
<img width="541" alt="image" src="https://github.com/platformio/platform-espressif32/assets/120377875/e71c3ca9-cf49-475e-bd7e-6f801179c09c">

Starting of the espota.py file
<img width="596" alt="image" src="https://github.com/platformio/platform-espressif32/assets/120377875/eebdce11-0eb3-4cd6-ba02-8b2da634777c">

The whole espota.py and esptool.py are there in the zip

Please suggest something that I can try to solve this issue.

**Comment by meakashrao** [comment]
This is my platformio.ini file 
<img width="552" alt="image" src="https://github.com/platformio/platform-espressif32/assets/120377875/4c28f9c1-e96f-4476-a326-3d447e4c658f">

and the .cpp file for LITLLEFS_Config

#include "Arduino.h"
#include "main.h"
#include "LittleFS.h"  
#include "LITTLEFS_Config.h"
#include "Rotary_Encoder.h"
#include "Weight_Handler.h"
#include "FS.h"

// Function declarations
void checkLITTLEFSContent();              // Initializes LittleFS and reads the contents of the file
void listDir(fs::FS &fs, const char * dirname, uint8_t levels);
void readFile(fs::FS &fs, const char * path);
void readConfigFile(fs::FS &fs, const char * path);

void checkLITTLEFSContent()
{
    Serial.println("Mounting LittleFS...");
    if (!LittleFS.begin(true))
    {
        Serial.println("An Error has occurred while mounting LittleFS");
        //TODO: Error Handling
        errorType = 2;
        addErrorMessage("MountingLITLFSError");
        return;
    }

    listDir(LittleFS, "/", 0);

    String configFile = (machineVersion == "BB1") ? "/BB1_config.txt" : "/BB2_config.txt";

    readFile(LittleFS, configFile.c_str()); // Use c_str() to convert String to const char*
    readConfigFile(LittleFS, configFile.c_str());
}

void readFile(fs::FS &fs, const char * path)
{
  Serial.printf("Reading file: %s\r\n", path);

  File file = fs.open(path);
  if(!file || file.isDirectory())
  {
    Serial.println("- failed to open file for reading");
    addErrorMessage("LITLFS-FileReadFail");
    return;
  }

  Serial.println("File Content:");
  while(file.available())
  {
    Serial.write(file.read());
  }
  file.close();
}

// Function to list the contents of a directory
void listDir(fs::FS &fs, const char * dirname, uint8_t levels) 
{
    Serial.printf("Listing directory: %s\n", dirname);

    File root = fs.open(dirname);
    if (!root) 
    {
        Serial.println("- failed to open directory");
        //TODO: Error Handling
        errorType = 4;  
        addErrorMessage("Directory-openfaile");
        return;
    }
    if (!root.isDirectory()) 
    {
        Serial.println(" - not a directory");
        //TODO: Error Handling
        errorType = 5;  
        addErrorMessage("Not-a-directory----");
        return;
    }

    const char* fileNames[] = {"BB1_config.txt", "BB2_config.txt"};         // Define the file names to check in the directory
    bool foundFiles[] = {false, false};                                     // Corresponds to the files in fileNames
    int numberOfFiles = sizeof(fileNames) / sizeof(char*);

    File file = root.openNextFile();
    while (file) 
    {
        Serial.print(file.isDirectory() ? "  DIR : " : "  FILE: ");
        Serial.println(file.name());
        Serial.print(file.size());                              // Print the size of the file
        Serial.println(" bytes");

        for (int i = 0; i < numberOfFiles; i++) 
        {
            if (String(file.name()).endsWith(fileNames[i]))     // Check if the file name ends with the desired file name
            { 
                Serial.printf("Found: %s\n", file.name());
                foundFiles[i] = true; // Mark as found
            }
        }
        if (file.isDirectory() && levels) 
        {
            listDir(fs, file.name(), levels - 1);               // Recursively list subdirectories
        }
        file = root.openNextFile();
    }
    // Check which files were not found and print the messages after listing
    if (foundFiles[0] == false) 
    {
        Serial.printf("Not found: %s\n", fileNames[0]);
        //TODO: Error Handling
        errorType = 6;      
        addErrorMessage("BB1-config-notfound");                            
    }
    if (foundFiles[1] == false) 
    {
        Serial.printf("Not found: %s\n", fileNames[1]);
        //TODO: Error Handling
        errorType = 7;   
        addErrorMessage("BB2-config-notfound");                               
    }
}

void readConfigFile(fs::FS &fs, const char * path) 
{
    Serial.printf("Reading file: %s\n", path);

    File file = fs.open(path);
    if (!file || file.isDirectory()) 
    {
        Serial.println("- failed to open file for reading");
        //TODO: Error Handling
        errorType = 3;
        addErrorMessage("LITLFS2FileReadFail");
        return;
    }
    while (file.available()) 
    {
        String line = file.readStringUntil('\n');   // Read a line of text up to the newline character.
        line.trim();                                // Remove any leading or trailing whitespace from the line.
        int equalSignIndex = line.indexOf('=');     // Find the position of the equal sign, which separates the key from the value.
        if (equalSignIndex != -1)                   // If there is an equal sign, then it's a line with a setting to process.
        {                 
            String key = line.substring(0, equalSignIndex);     // Extract the key (the part of the line before the equal sign).
            key.trim();                                         // Remove any whitespace around the key.
            String valueStr = line.substring(equalSignIndex + 1);   // Extract the value (the part of the line after the equal sign).
            valueStr.trim();                                    // Remove any whitespace around the value.
            // Call the appropriate function to handle the setting based on the key.
            // These functions will parse the value string and set the corresponding configuration variables.
            readRGBPins(valueStr, key);
            readHX711(valueStr, key);
            //readQTempTrigger(valueStr, key);
            readRGBArray(valueStr, key);
        }
    }
    file.close();  // After reading all the settings, close the file.
}


Is there something wrong with this?? I think that after OTA, the files are not written to the correct location, how to set the correct offset while trying to OTA the littlefs file?

**Comment by meakashrao** [comment]
I am trying to update the littlefs.bin file to a SPIFFS partition (min_spiffs.csv), and I am using Platform IO in VS Code with Core 6.1.13 and Home 3.4.4 with Espressif 32 v6.4.0

When I do a serial update of firmware.bin and littlefs.bin, things work well and I am able to read all data successfully, but when I do an OTA update using espota.py, 
a) after updating firmware.bin I get failure message of "corrupted dir pair at {0x0, 0x1}, mount failed, (-84) Failed to initialize LittleFS" and since I have set LittleFS.begin(true) after a power on reset, LittleFs mount is successful
b) after updating the littlefs.bin I get an error message "Invalid block count (32 != 48) mount failed, (-22) Failed to initialize LittleFS"

I thought this is something to do with espota.py not being able to write files at the correct location, so I used esptool.py and downloaded the flash content using flash read first with only OTA update and then use the same machine to serially upload both files and read the flash again.
I saw the bin files were writing spiffs data to correct locations. 


I have attached a zip folder which contains 
1) ota_flash-machine2_dump.bin -> Bin file read from flash after OTA update of both firmware and littlefs files
2) serial_flash-machine2_dump.bin -> Bin file read from flash after serial flashing of both firmware and littlefs files
3) Serial and OTA update.docx -> this contains all screenshots of Serial terminal and the screenshots of above bin files
4) firmware.bin -> This is the actual firmware.bin being used for OTA and serial update
5) littlefs.bin -> This is the actual firmware.bin being used for OTA and serial update
6) LITTLEFS_Config.cpp -> This is my LittleFS config file
7) platformio_ini -> PlatformIO configuration File
8) espota.py -> file that I am using
9) esptool.py -> file that I am using 

[LittleFs_Debug.zip](https://github.com/platformio/platform-espressif32/files/13974198/LittleFs_Debug.zip)

**Comment by meakashrao** [comment]
Hello All,

I found the problem about this and why wasn't it working from OTA and was working with serially.

The development first happened with Arduino IDE and min_spiffs.csv was used and was modified to not include coredump and the extra space was added to SPIFFS
When I was trying to Platform IO, I didn't modify this core dump partition and that is why the error happened.

After modifying the coredump size in VS Code partition tables, I was able to make it work through OTA.

I am using Espressif 32-6.4.0 with Platform IO core 6.1.13 and Home 3.4.4 and Littlefs.bin upload through espota.py does work.

Ps: I am using an older version of espota.py, I am attaching it below for reference.
[espota.zip](https://github.com/platformio/platform-espressif32/files/14007658/espota.zip)

Thank you once again and sorry for the trouble
